src/main/java/oxgame/app/GameInProgress.java

```diff
@@ -0,0 +1,57 @@
+package oxgame.app;
+
+import java.util.function.Consumer;
+
+public class GameInProgress implements GameState {
+
+    private final RoundBuffer playerBuffer;
+    private final Consumer<String> output;
+    private Board board;
+    private Player player;
+    private int currentBoardSize;
+
+    public GameInProgress(RoundBuffer playerBuffer, Consumer<String> output, Board board) {
+        this.playerBuffer = playerBuffer;
+        this.output = output;
+        this.board = board;
+        currentBoardSize = board.size();
+    }
+
+
+    @Override
+    public void showState() {
+        player = playerBuffer.takePlayer();
+        output.accept(player.toString());
+        board.showBoard();
+    }
+
+    @Override
+    public GameState nextState(String input) throws IllegalArgumentException{
```

Comment:
This method is really long, try to refactor all of the if statements into a separate method